diff --git a/src/gigedit/CombineInstrumentsDialog.cpp b/src/gigedit/CombineInstrumentsDialog.cpp
index a3029e0..230122a 100644
--- a/src/gigedit/CombineInstrumentsDialog.cpp
+++ b/src/gigedit/CombineInstrumentsDialog.cpp
@@ -941,7 +941,7 @@ CombineInstrumentsDialog::CombineInstrumentsDialog(Gtk::Window& parent, gig::Fil
     }
 
     m_buttonBox.set_layout(Gtk::BUTTONBOX_END);
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     m_buttonBox.set_margin(5);
 #else
     m_buttonBox.set_border_width(5);
diff --git a/src/gigedit/MacroEditor.cpp b/src/gigedit/MacroEditor.cpp
index 75277bf..104a6a1 100644
--- a/src/gigedit/MacroEditor.cpp
+++ b/src/gigedit/MacroEditor.cpp
@@ -232,7 +232,7 @@ void MacroEditor::buildTreeView(const Gtk::TreeModel::Row& parentRow, const Seri
         const Serialization::Member& member = parentObject.members()[iMember];
         const Serialization::Object& object = m_macro.objectByUID(member.uid());
         
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
         //HACK: on GTKMM 3.9x append() below requires TreeNodeChildren, parentRow.children() returns TreeNodeConstChildren though, probably going to be fixed before final GTKMM4 release though.
         const Gtk::TreeNodeConstChildren& children = parentRow.children();
         Gtk::TreeNodeChildren* const pChildren = (Gtk::TreeNodeChildren* const) &children;
diff --git a/src/gigedit/ReferencesView.cpp b/src/gigedit/ReferencesView.cpp
index 01746fa..6f4e250 100644
--- a/src/gigedit/ReferencesView.cpp
+++ b/src/gigedit/ReferencesView.cpp
@@ -66,7 +66,7 @@ ReferencesView::ReferencesView(Gtk::Window& parent) :
     );
 
     m_buttonBox.set_layout(Gtk::BUTTONBOX_END);
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     m_buttonBox.set_margin(5);
 #else
     m_buttonBox.set_border_width(5);
diff --git a/src/gigedit/compat.h b/src/gigedit/compat.h
index 5d34a49..cfb99f2 100644
--- a/src/gigedit/compat.h
+++ b/src/gigedit/compat.h
@@ -61,7 +61,7 @@
 // Gtk::UIManager had been replaced by Gtk::Builder in GTKMM 3 and removed in GTKMM 4
 // Gtk::ActionGroup had been replaced by Gio::ActionGroup in GTKMM 3 and removed in GTKMM 4
 #if !defined(USE_GTKMM_BUILDER) || !defined(USE_GLIB_ACTION) // gtkmm > 3.22 :
-# if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+# if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
 #  define USE_GTKMM_BUILDER 1
 #  define USE_GLIB_ACTION 1
 # else
@@ -79,7 +79,7 @@
 #endif
 
 #ifndef HAS_GTKMM_CPP11_ENUMS // gtkmm > 3.22
-# if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+# if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
 #  define HAS_GTKMM_CPP11_ENUMS 1
 # else
 #  define HAS_GTKMM_CPP11_ENUMS 0
@@ -88,7 +88,7 @@
 
 // alignment.h had been removed in GTKMM 4
 #ifndef HAS_GTKMM_ALIGNMENT // gtkmm > 3.22 :
-# if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+# if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
 #  define HAS_GTKMM_ALIGNMENT 0
 # else
 #  define HAS_GTKMM_ALIGNMENT 1
@@ -99,7 +99,7 @@
 // stock.h had been removed in GTKMM 4
 // Gtk::VBox, Gtk::HBox and Gtk::HButtonBox had been replaced by Gtk::Box in GTKMM 3 and removed in GTKMM 4
 #if !defined(USE_GTKMM_GRID) || !defined(HAS_GTKMM_STOCK) || !defined(USE_GTKMM_BOX) || !defined(USE_GTKMM_PANED) // gtkmm > 3.22 :
-# if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+# if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
 #  define USE_GTKMM_GRID 1
 #  define HAS_GTKMM_STOCK 0
 #  define USE_GTKMM_BOX 1
diff --git a/src/gigedit/dimensionmanager.cpp b/src/gigedit/dimensionmanager.cpp
index f585c87..3be4d7c 100644
--- a/src/gigedit/dimensionmanager.cpp
+++ b/src/gigedit/dimensionmanager.cpp
@@ -278,7 +278,7 @@ DimensionManager::DimensionManager() :
     scrolledWindow.show();
     vbox.pack_start(buttonBox, Gtk::PACK_SHRINK);
     buttonBox.set_layout(Gtk::BUTTONBOX_END);
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     buttonBox.set_margin(5);
 #else
     buttonBox.set_border_width(5);
diff --git a/src/gigedit/dimregionchooser.cpp b/src/gigedit/dimregionchooser.cpp
index 114bb1a..350aeda 100644
--- a/src/gigedit/dimregionchooser.cpp
+++ b/src/gigedit/dimregionchooser.cpp
@@ -289,7 +289,7 @@ DimRegionChooser::DimRegionChooser(Gtk::Window& window) :
 #endif // USE_GTKMM_BUILDER
 
 
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
 # warning GTKMM4 event registration code missing for dimregionchooser!
     //add_events(Gdk::EventMask::BUTTON_PRESS_MASK);
 #else
diff --git a/src/gigedit/mainwindow.cpp b/src/gigedit/mainwindow.cpp
index d43436e..92416a9 100644
--- a/src/gigedit/mainwindow.cpp
+++ b/src/gigedit/mainwindow.cpp
@@ -2765,7 +2765,7 @@ PropDialog::PropDialog()
 #endif
 
     add(vbox);
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     table.set_margin(5);
 #else
     table.set_border_width(5);
@@ -2773,7 +2773,7 @@ PropDialog::PropDialog()
     vbox.add(table);
     vbox.pack_start(buttonBox, Gtk::PACK_SHRINK);
     buttonBox.set_layout(Gtk::BUTTONBOX_END);
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     buttonBox.set_margin(5);
 #else
     buttonBox.set_border_width(5);
@@ -2925,7 +2925,7 @@ InstrumentProps::InstrumentProps() :
     table.add(eDimensionKeyRangeHigh);
 
     add(vbox);
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     table.set_margin(5);
 #else
     table.set_border_width(5);
@@ -2934,7 +2934,7 @@ InstrumentProps::InstrumentProps() :
     table.show();
     vbox.pack_start(buttonBox, Gtk::PACK_SHRINK);
     buttonBox.set_layout(Gtk::BUTTONBOX_END);
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     buttonBox.set_margin(5);
 #else
     buttonBox.set_border_width(5);
@@ -3453,7 +3453,7 @@ void MainWindow::select_instrument(gig::Instrument* instrument) {
             show_intruments_tab();
             m_TreeView.get_selection()->unselect_all();
             
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
             auto iterSel = model->children()[i].get_iter();
             m_TreeView.get_selection()->select(iterSel);
 #else
@@ -3483,7 +3483,7 @@ bool MainWindow::select_dimension_region(gig::DimensionRegion* dimRgn) {
             // select and show the respective instrument in the list view
             show_intruments_tab();
             m_TreeView.get_selection()->unselect_all();
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
             auto iterSel = model->children()[i].get_iter();
             m_TreeView.get_selection()->select(iterSel);
 #else
@@ -3519,7 +3519,7 @@ void MainWindow::select_sample(gig::Sample* sample) {
             if (rowSample[m_SamplesModel.m_col_sample] == sample) {
                 show_samples_tab();
                 m_TreeViewSamples.get_selection()->unselect_all();
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
                 auto iterSel = rowGroup.children()[s].get_iter();
                 m_TreeViewSamples.get_selection()->select(iterSel);
 #else
@@ -4794,7 +4794,7 @@ bool MainWindow::instrument_row_visible(const Gtk::TreeModel::const_iterator& it
     trim(pattern);
     if (pattern.empty()) return true;
 
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     //HACK: on GTKMM4 development branch const_iterator cannot be easily converted to iterator, probably going to be fixed before final GTKMM4 release though.
     Gtk::TreeModel::Row row = **(Gtk::TreeModel::iterator*)(&iter);
 #else
diff --git a/src/gigedit/midirules.cpp b/src/gigedit/midirules.cpp
index 0f1b041..aee093e 100644
--- a/src/gigedit/midirules.cpp
+++ b/src/gigedit/midirules.cpp
@@ -40,7 +40,7 @@ MidiRules::MidiRules() :
     }
 
     set_title(_("Midi Rules"));
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     set_margin(6);
 #else
     set_border_width(6);
@@ -48,7 +48,7 @@ MidiRules::MidiRules() :
 
     add(vbox);
 
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     hbox.set_margin(6);
 #else
     hbox.set_border_width(6);
@@ -69,14 +69,14 @@ MidiRules::MidiRules() :
         sigc::mem_fun(*this, &MidiRules::combo_changed));
     vbox.pack_start(hbox, Gtk::PACK_SHRINK);
 
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     box.set_margin(6);
 #else
     box.set_border_width(6);
 #endif
     vbox.pack_start(box);
 
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     button_box.set_margin(6);
 #else
     button_box.set_border_width(6);
diff --git a/src/gigedit/regionchooser.cpp b/src/gigedit/regionchooser.cpp
index 2147bc0..399e230 100644
--- a/src/gigedit/regionchooser.cpp
+++ b/src/gigedit/regionchooser.cpp
@@ -269,7 +269,7 @@ RegionChooser::RegionChooser() :
 
 #endif // USE_GTKMM_BUILDER
 
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
 # warning GTKMM4 event registration code missing for regionchooser!
     //add_events(Gdk::EventMask::BUTTON_PRESS_MASK);
 #else
@@ -366,7 +366,7 @@ bool RegionChooser::on_draw(const Cairo::RefPtr<Cairo::Context>& cr) {
 #if (GTKMM_MAJOR_VERSION == 2 && GTKMM_MINOR_VERSION < 90) || GTKMM_MAJOR_VERSION < 2
     const Gdk::Color bg = get_style()->get_bg(Gtk::STATE_NORMAL);
 #else
-#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 22)
+#if GTKMM_MAJOR_VERSION > 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION > 24)
     GdkRGBA gdkBgRGBA;
     gtk_style_context_get_background_color(get_style_context()->gobj(), &gdkBgRGBA);
     const Gdk::RGBA bg = Glib::wrap(&gdkBgRGBA, true);
diff --git a/src/gigedit/scripteditor.cpp b/src/gigedit/scripteditor.cpp
index 3ce3a12..30dcc12 100644
--- a/src/gigedit/scripteditor.cpp
+++ b/src/gigedit/scripteditor.cpp
@@ -265,7 +265,7 @@ ScriptEditor::ScriptEditor() :
 #endif
         color.set("#F5F5F5");
         GtkWidget* widget = (GtkWidget*) m_lineNrView.gobj();
-#if GTK_MAJOR_VERSION < 3 || (GTK_MAJOR_VERSION == 3 && GTK_MINOR_VERSION <= 22)
+#if GTK_MAJOR_VERSION < 3 || (GTK_MAJOR_VERSION == 3 && GTK_MINOR_VERSION <= 24)
         gtk_widget_modify_base(widget, GTK_STATE_NORMAL, color.gobj());
         gtk_widget_modify_bg(widget, GTK_STATE_NORMAL, color.gobj());
 #endif
@@ -278,7 +278,7 @@ ScriptEditor::ScriptEditor() :
 #endif
         color.set("#EEEEEE");
         GtkWidget* widget = (GtkWidget*) m_lineNrTextViewSpacer.gobj();
-#if GTK_MAJOR_VERSION < 3 || (GTK_MAJOR_VERSION == 3 && GTK_MINOR_VERSION <= 22)
+#if GTK_MAJOR_VERSION < 3 || (GTK_MAJOR_VERSION == 3 && GTK_MINOR_VERSION <= 24)
         gtk_widget_modify_base(widget, GTK_STATE_NORMAL, color.gobj());
         gtk_widget_modify_bg(widget, GTK_STATE_NORMAL, color.gobj());
 #endif
@@ -759,7 +759,7 @@ bool ScriptEditor::on_motion_notify_event(GdkEventMotion* e) {
     //TODO: event throttling would be a good idea here
     updateIssueTooltip(e);
 #endif
-#if GTKMM_MAJOR_VERSION < 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION <= 22)
+#if GTKMM_MAJOR_VERSION < 3 || (GTKMM_MAJOR_VERSION == 3 && GTKMM_MINOR_VERSION <= 24)
     return ManagedWindow::on_motion_notify_event(e);
 #else
     Gdk::EventMotion em = Glib::wrap(e, true);
