diff --git a/src/app/app.pro b/src/app/app.pro
index 2f28858..2362e34 100644
--- a/src/app/app.pro
+++ b/src/app/app.pro
@@ -338,7 +338,7 @@ for(_translation_name, INSTALL_TRANSLATIONS) {
   _translation_name_qm = $$basename(_translation_name)
   _translation_name_ts = $$section(_translation_name_qm, ".", 0, 0).ts
 
-  system($$shell_path($$[QT_INSTALL_BINS]/$$LRELEASE) -removeidentical -nounfinished $$shell_path($${PWD}/$${TRANSLATIONS_PATH}/$$_translation_name_ts) -qm $$shell_path($${PWD}/$$_translation_name))
+  system($$LRELEASE -removeidentical -nounfinished $$shell_path($${PWD}/$${TRANSLATIONS_PATH}/$$_translation_name_ts) -qm $$shell_path($${PWD}/$$_translation_name))
 }
 
 for(DIR, INSTALL_TRANSLATIONS) {
