diff --git a/pgfouine-1.2/include/postgresql/lines/PostgreSQLHintLine.class.php b/pgfouine-1.2/include/postgresql/lines/PostgreSQLHintLine.class.php
index 274f238..f23debd 100644
--- a/pgfouine-1.2/include/postgresql/lines/PostgreSQLHintLine.class.php
+++ b/pgfouine-1.2/include/postgresql/lines/PostgreSQLHintLine.class.php
@@ -25,7 +25,9 @@ class PostgreSQLHintLine extends PostgreSQLLogLine {
 	var $ignore = false;
 
 	function appendTo(& $logObject) {
-		$logObject->appendHint($this->text);
+		if (method_exists($logObject, 'appendHint')) {
+			$logObject->appendHint($this->text);
+		}
 	}
 	
 	function isContextual() {
@@ -33,4 +35,4 @@ class PostgreSQLHintLine extends PostgreSQLLogLine {
 	}
 }
 
-?>
\ No newline at end of file
+?>
