diff --git a/include/CsvlogLogReader.class.php b/include/CsvlogLogReader.class.php
index cbd18df..e4a5744 100644
--- a/include/CsvlogLogReader.class.php
+++ b/include/CsvlogLogReader.class.php
@@ -32,7 +32,7 @@ class CsvlogLogReader extends GenericLogReader {
 		while ($csvLine = fgetcsv($filePointer)) {
 			$lineParsedCounter ++;
 			
-			if(count($csvLine) == 22) {
+			if(count($csvLine) == 22 || (count($csvLine) == 23) {
 				$lines =& $lineParser->parse($csvLine);
 				
 				if($lines) {
